### PR TITLE
Enable AudioDSP

### PR DESCRIPTION
**Pull request by AchimTuran**
This PR enables AudioDSP.
It should work, but it needs further testing under Linux. I know it is not the best way, but I need something to play with and I try to PR and merge as much as possible. Further improvements will follow in my next PRs. Before we merge this we should merge https://github.com/FernetMenta/kodi-agile/pull/24.

TODO:
- test with add-ons
- test with Linux

### CODE REVIEW

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp [hunk 1]
```diff
@@ -598,20 +601,26 @@ bool CActiveAEStreamBuffers::HasInputLevel(int level)
...
+{  
+  m_UseADSP = useDSP;
+
   if (!m_resampleBuffers->Create(totaltime, remap, upmix, normalize))
     return false;
 
   if (!m_atempoBuffers->Create(totaltime))
     return false;
 
+  if (!m_adspBuffers->Create(totaltime, upmix, bypassDSP))
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp]: I don't like tis logic here. If we don't use or bypass adsp, we should not create the instance

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp [hunk 2]
```diff
@@ -650,11 +680,18 @@ bool CActiveAEStreamBuffers::ProcessBuffers()
...
-void CActiveAEStreamBuffers::ConfigureResampler(bool normalizelevels, bool dspenabled, bool stereoupmix, AEQuality quality)
+void CActiveAEStreamBuffers::ConfigureResampler(bool normalizelevels, bool stereoupmix, AEQuality quality)
 {
   m_resampleBuffers->ConfigureResampler(normalizelevels, stereoupmix, quality);
 }
 
+void ActiveAE::CActiveAEStreamBuffers::ConfigureADSP(bool useDSP, bool stereoupmix, AEQuality quality)
+{
+  /*! @todo pass bypass, quality, upmix to AudioDSP*/
+  m_UseADSP = useDSP;
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp]: this makes no sense. if it is not supposed to be used, it should not be configured

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 3]
```diff
@@ -466,6 +467,7 @@ void CActiveAE::StateMachine(int signal, Protocol *port, Message *msg)
           m_extError = false;
           m_sink.EnumerateSinkList(false);
           LoadSettings();
+          CServiceBroker::GetADSP().Init();
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: this should not be called if ADSP is not used

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp]: bypass is dead code anyway and should be removed for now. I never ever add dead code

**Comment by agileJames** [comment]
Is rebased. Currently trying the build under Linux.

**Pull request closed without merging**